cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/ViewStatusClient.scala

```diff
@@ -0,0 +1,108 @@
+package edu.uci.ics.cloudberry.zion.actor
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props, Stash}
+import akka.pattern.ask
+import akka.util.Timeout
+import edu.uci.ics.cloudberry.zion.actor.DataStoreManager.{AskInfo, AskInfoAndViews}
+import edu.uci.ics.cloudberry.zion.common.Config
+import edu.uci.ics.cloudberry.zion.model.datastore.{IPostTransform}
+import edu.uci.ics.cloudberry.zion.model.impl.{DataSetInfo, JSONParser, QueryPlanner}
+import edu.uci.ics.cloudberry.zion.model.schema.Query
+import play.api.libs.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * A reactive client which checks whether a query can be solved by the existed view.
+  *
+  */
+class ViewStatusClient(val jsonParser: JSONParser,
+                  val dataManager: ActorRef,
+                  val planner: QueryPlanner,
+                  val config: Config,
+                  val out: ActorRef
+                 )(implicit val ec: ExecutionContext) extends Actor with Stash with ActorLogging {
+
+  import ViewStatusClient._
+
+  // Timeout due to the configuration
+  implicit val askTimeOut: Timeout = config.UserTimeOut
+
+  // Handle the request when the client received message
+  override def receive: Receive = {
+    case json: JsValue =>
+      handleRequest(json)
+    case (json: JsValue, transform: IPostTransform) =>
+      handleRequest(json)
+  }
+
+  // Handle the Request in following step
+  private def handleRequest(json: JsValue): Unit = {
+    // 1. Use jsonParser to parse the json to get datasets name and queryID
+    val datasets = jsonParser.getDatasets(json).toSeq
+    val queryID = jsonParser.getQueryID(json)
```

Comment:
About queryID part, Cloudberry query format has already has an ID parameter in `transform` object, (refer to https://github.com/ISG-ICS/cloudberry/pull/471/files). So we can reuse that as the queryID functionality.